wallets/server.js

```diff
@@ -41,90 +42,125 @@ export async function createInvoice (userId, { msats, description, descriptionHa
 
   for (const wallet of wallets) {
     const w = walletDefs.find(w => w.walletType === wallet.type)
-    try {
-      if (!canReceive({ def: w, config: wallet.wallet })) {
-        continue
-      }
 
-      const { walletType, walletField, createInvoice } = w
+    const config = wallet.wallet
+    if (!canReceive({ def: w, config })) {
+      continue
+    }
 
-      const walletFull = await models.wallet.findFirst({
-        where: {
-          userId,
-          type: walletType
-        },
-        include: {
-          [walletField]: true
-        }
-      })
+    const logger = walletLogger({ wallet, models })
 
-      if (!walletFull || !walletFull[walletField]) {
-        throw new Error(`no ${walletType} wallet found`)
+    try {
+      logger.info(
+        `↙ incoming payment: ${formatSats(msatsToSats(msats))}`,
+        {
+          amount: formatMsats(msats)
+        })
+
+      let invoice
+      try {
+        invoice = await walletCreateInvoice(
+          { msats, description, descriptionHash, expiry },
+          { ...w, userId, createInvoice: w.createInvoice },
+          { logger, models })
+      } catch (err) {
+        throw new Error('failed to create invoice: ' + err.message)
       }
 
-      // check for pending withdrawals
-      const pendingWithdrawals = await models.withdrawl.count({
-        where: {
-          walletId: walletFull.id,
-          status: null
-        }
-      })
+      const bolt11 = await parsePaymentRequest({ request: invoice })
 
-      // and pending forwards
-      const pendingForwards = await models.invoiceForward.count({
-        where: {
-          walletId: walletFull.id,
-          invoice: {
-            actionState: {
-              notIn: PAID_ACTION_TERMINAL_STATES
-            }
-          }
-        }
+      logger.info(`created invoice for ${formatSats(msatsToSats(bolt11.mtokens))}`, {
+        bolt11: invoice
       })
 
-      console.log('pending invoices', pendingWithdrawals + pendingForwards)
-      if (pendingWithdrawals + pendingForwards >= MAX_PENDING_INVOICES_PER_WALLET) {
-        throw new Error('wallet has too many pending invoices')
-      }
-      console.log('use wallet', walletType)
-
-      const invoice = await withTimeout(
-        createInvoice({
-          msats,
-          description: wallet.user.hideInvoiceDesc ? undefined : description,
-          descriptionHash,
-          expiry
-        }, walletFull[walletField]), 10_000)
-
-      const bolt11 = await parsePaymentRequest({ request: invoice })
       if (BigInt(bolt11.mtokens) !== BigInt(msats)) {
         if (BigInt(bolt11.mtokens) > BigInt(msats)) {
-          throw new Error(`invoice is for an amount greater than requested ${bolt11.mtokens} > ${msats}`)
+          throw new Error('invoice invalid: amount too big')
```

Comment:
we don't need to add specific info into the log message anymore since the exact values can be derived from the context now